pkg/ebpf/verifier/stats.go

```diff
@@ -238,6 +239,14 @@ func generateLoadFunction(file string, opts *StatsOptions, results *StatsResult,
 					return fmt.Errorf("Unexpected type %T", field)
 				}
 			}
+
+			// After each program, force Go to release as much memory as possible
+			// With line-complexity enabled, each program allocates a 1GB buffer for the
+			// verifier log, which means that the memory footprint of the program can get
+			// quite large before the garbage collector kicks in and releases memory to the OS.
+			// This causes out-of-memory errors in CI specially, which an environment with higher memory
+			// restrictions and multiple programs running in different VMs.
+			debug.FreeOSMemory()
```

Comment:
the 1gb is accounted by the VerifierLog field of the Program struct?

in that case, maybe it is better to set explicitly the string to empty string, since the program parent struct is a held by pointer and it might cause the GC to skip freeing that memory (or delay the actual free operation)

Reply:
You may need to do both set the string to `""` and call `debug.FreeOSMemory()`

Reply:
I tested it with both ways and setting the string to `""` was not enough to make the GC collect the memory, that's why I added `debug.FreeOSMemory()`. Just in case I changed the code to set the verifier log to an empty string to ensure it's freed, although I think that `prog` being scoped to just the loop is enough for the GC to mark that memory as unused.

Reply:
golang GC works a bit differently when it comes to the pointers and heap memory